### PR TITLE
CAT-93-Add-possibility-to-modify-a-Konnektor-that-was-already-created

### PR DESCRIPTION
**Pull request by TimonWag**

### CODE REVIEW

File: src/main/webapp/dashboard/controller/Master.controller.js [hunk 1]
```diff
@@ -66,10 +68,65 @@ sap.ui.define(
           oBinding.sort(oSorter);
         },
 
+        onChange: function(){
+           const oView = this.getView();
+           const aSelectedItems = oView.byId("runtimeConfigTable").getSelectedItems();
+           if(aSelectedItems.length == 1){
```
Comment (BeatrizRCorreia) [inline src/main/webapp/dashboard/controller/Master.controller.js:74]: Please replace loose equality ("==") with strict equality ("===") for improved type checking.

File: src/main/webapp/dashboard/controller/Master.controller.js [hunk 2]
```diff
@@ -66,10 +68,65 @@ sap.ui.define(
...
+                       }).then(function (oDialog) {
+                         oView.addDependent(oDialog);
+                         me._openChangeDialog(oDialog);
+                       });
+                     } else {
+                       this._openChangeDialog(this.byId("changeDialog"));
+                     }
+           }
+           else{
+            if(aSelectedItems.length == 0){
```
Comment (BeatrizRCorreia) [inline src/main/webapp/dashboard/controller/Master.controller.js:91]: Please replace loose equality ("==") with strict equality ("===") for improved type checking.

File: src/main/webapp/dashboard/controller/Master.controller.js [hunk 3]
```diff
@@ -66,10 +68,65 @@ sap.ui.define(
...
+                MessageBox.error("Bitte wählen Sie einen Konnektor aus.");
+            }
+            else if(aSelectedItems.length > 1){
+                MessageBox.error("Bitte wählen Sie nur einen Konnektor aus.");
+            }
+           }
+        },
+
+        _openChangeDialog: function (oDialog, sEntityName) {
```
Comment (BeatrizRCorreia) [inline src/main/webapp/dashboard/controller/Master.controller.js:100]: Please remove sEntityName parameter as it is not being used.

File: src/main/webapp/dashboard/controller/Master.controller.js [hunk 4]
```diff
@@ -66,10 +68,65 @@ sap.ui.define(
...
+                     } else {
+                       this._openChangeDialog(this.byId("changeDialog"));
+                     }
+           }
+           else{
+            if(aSelectedItems.length == 0){
+                MessageBox.error("Bitte wählen Sie einen Konnektor aus.");
+            }
+            else if(aSelectedItems.length > 1){
+                MessageBox.error("Bitte wählen Sie nur einen Konnektor aus.");
```
Comment (BeatrizRCorreia) [inline src/main/webapp/dashboard/controller/Master.controller.js:95]: Please add all the messages to the i18n file and reference them here like this:
MessageBox.error(this.translate("msgSelectOnlyOneConnector"));

File: src/main/webapp/dashboard/view/ChangeDialog.fragment.xml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+                <Input type="Text" value="{VzdPort}" />
+                <Label text="Mandant-ID" />
+                <Input type="Text" value="{MandantId}" />
+                <Label text="System-ID Kunde" />
+                <Input type="Text" value="{ClientSystemId}" />
+                <Label text="Arbeitsplatz-ID" />
+                <Input type="Text" value="{WorkplaceId}" />
+                <Label text="Kundenzertifikat" />
+                <Input type="Text" value="{ClientCertificate}" />
+                <Label text="Kennwort Kundenzertifikat" />
```
Comment (BeatrizRCorreia) [inline src/main/webapp/dashboard/view/ChangeDialog.fragment.xml:25]: Please reference these words from the i18n file.

File: src/main/webapp/dashboard/view/Master.view.xml [hunk 6]
```diff
@@ -33,6 +33,8 @@
                                                press="onAdd" tooltip="{i18n>AddNewConnector}"/>
                         <OverflowToolbarButton icon="sap-icon://sort" type="Transparent"
                                                press="onSort" tooltip="{i18n>Sort}"/>
+                        <OverflowToolbarButton icon="sap-icon://edit" type="Transparent"
+                                               press="onChange" tooltip="{i18n>Change}"/>
```
Comment (BeatrizRCorreia) [inline src/main/webapp/dashboard/view/Master.view.xml:37]: Please also rename "Change" to "Edit" everywhere.

Comment (ManuelB) [reply src/main/webapp/dashboard/view/Master.view.xml:37]: In general we should try to follow the SAP Fiori Design guidelines:
https://experience.sap.com/fiori-design-web/object-page/

So like Beatriz said they also call it edit:
https://experience.sap.com/fiori-design-web/object-page/#global-edit

**Pull request merged**